packages/core/src/shared/ui/nestedWizardPrompter.ts

```diff
@@ -0,0 +1,63 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Wizard, WizardOptions } from '../wizards/wizard'
+import { Prompter } from './prompter'
+import { WizardPrompter } from './wizardPrompter'
+import { createHash } from 'crypto'
+
+/**
+ * An abstract class that extends the base Wizard class plus the ability to
+ * use other wizard classes as prompters
+ */
+export abstract class NestedWizard<T> extends Wizard<T> {
+    /**
+     * Map to store memoized wizard instances using SHA-256 hashed keys
+     */
+    private wizardInstances: Map<string, any> = new Map()
```

Comment:
Usually not necessary to mention "instances" in variable names. Is there something more meaningful, maybe `childWizards` (I'm not sure if that's correct)?

```suggestion
    private childWizards: Map<string, any> = new Map()
```

Reply:
Updated.